azure-core/README.md

```diff
@@ -0,0 +1,357 @@
+```
+
+### Configuration
+
+The Configuration object is the home of all the configurable policies in the pipeline. It is also used
+to provide configuration parameters for the Transport. A new Configuration object provides *no default policies*.
+It is up to the SDK developer to specify each of the policy defaults as required by the service.
+
+This can be seen in the above code sample as implemented in a staticmethod on the client class.
+The Configuration object does not specify in what order the policies will be added to the pipeline.
+It is up to the SDK developer to use the policies in the Configuration to construct the pipeline correctly, as well
+as inserting any unexposed/non-configurable policies.
+```python
+config = config or FooServiceClient.create_config(**kwargs)
+transport = RequestsTransport(config)
+
+# SDK developer needs to build the policy order for the pipeline.
+policies = [
+    config.user_agent_policy,
+    config.headers_policy,
+    credentials,  # Credentials policy needs to be inserted after all request mutation to accomodate signing.
+    ContentDecodePolicy(),
+    config.redirect_policy,
+    config.retry_policy,
+    config.logging_policy,  # Logger should come last to accurately record the request/response as they are on the wire
+]
+self._pipeline = Pipeline(transport, policies=policies)
+```
+The policies that should currently be defined on the Configuration object are as follows:
+```python
+- Configuration.headers_policy  # HeadersPolicy
+- Configuration.retry_policy  # RetryPolicy
+- Configuration.redirect_policy  # RedirectPolicu
+- Configuration.logging_policy  # NetworkTraceLoggingPolicy
+- Configuration.user_agent_policy  # UserAgentPolicy
+- Configuration.connection  # The is a ConnectionConfiguration, used to provide common transport parameters.
+- Configuration.proxy_policy  # While this is a ProxyPolicy object, current implementation is transport configuration.
+```
+
+### Transport 
+
+Various combinations of sync/async HTTP libraries as well as alternative event loop implementations are available. Therefore to support the widest range of customer scenarios, we must allow a customer to easily swap out the HTTP transport layer to one of those supported.
+
+The transport is the last node in the pipeline, and adheres to the same basic API as any policy within the pipeline.
+The only currently available transport for synchronous pipelines uses the `Requests` library:
+```
+from azure.core.pipeline.transport import RequestsTransport
+synchronous_transport = RequestsTransport()
+```
+
+For asynchronous pipelines a couple of transport options are available. Each of these transports are interchangable depending on whether the user has installed various 3rd party dependencies (i.e. aiohttp or trio), and the user
+should easily be able to specify their chosen transport. SDK developers should use the `aiohttp` transport as the default for asynchronous pipelines where the user has not specified as alternative.
```

Comment:
We should probably do a scan to make sure we refer to things as libraries rather than SDK.

Reply:
Interesting.... so that will be the decided terminology?